tensorboard/plugins/hparams/api.proto

```diff
@@ -0,0 +1,346 @@
+  // is not ORDER_UNSPECIFIED. In each of these the order field specifies the
+  // order (ascending or descending) of the values in the associated column
+  // of the session group. The order of the ColParams messages themselves
+  // determines the "significance" of the associated column as a sorting key:
+  // with the first being the primary sorting key, the second being
+  // the secondary sorting key, etc.
+  repeated ColParams col_params = 1;
+
+  // Fields controlling how to aggregate metrics across sessions within a
+  // session group.
+  // If metric_aggregation_type is AGGREGATION_AVG, each metric value of the
+  // session group is the average of the values of the metric across the
+  // sessions (at the same computation step).
+  // Otherwise, each metric value of the session group is the value of that
+  // metric for a "representative" session in the group. A representative
+  // session is a session having a "representative" value for the metric
+  // specified by the 'primary_metric' field below. The representative value
+  // for a metric is its minimum, maximum or median value across the sessions
+  // in the group, for metric_aggregation_type being AGGREGATION_MIN,
+  // AGGREGATION_MAX, or AGGREGATION_MEDIAN, respectively.
+  AggregationType aggregation_type = 2;
+  MetricName primary_metric = 3;
+
+  // Pagination support:
+  // The number of session groups in each page to return.
+  int32 page_size = 4;
+  // The token representing the page to return. Should be empty for the first
+  // page and copied from the 'next_page_token' field of the response for
+  // later pages.
+  string page_token = 5;
+}
+
+// Defines parmeters for a ListSessionGroupsRequest for a specific column.
+message ColParams {
+  oneof name {
+    MetricName metric = 1;
+    string hparam = 2;
+  }
+
+  // Sorting.
+  SortOrder order = 3;
+
+  // Filtering. Defines a subset of the domain of the column.
+  ///Different representations for the subset are possible for different
+  // column data types (hyperparameter data types or DATA_TYPE_FLOAT64 for
+  // metrics). See the comments below for more
+  // details.
+  oneof filter {
+    // Only valid for string-valued hyperparameter columns. The subset is
+    // the set of all strings matching the regular expression stored
+    // in 'regexp'.
+    string regexp = 4;
+
+    // Only valid for numeric-valued columns. The subset is the given interval.
+    Interval interval = 5;
+
+    // Valid for all data types. The subset is defined explicitly.
+    google.protobuf.ListValue discrete_set = 6;
+  }
+}
+
+enum SortOrder {
+  ORDER_UNSPECIFIED = 0;
+  ORDER_ASC = 1;
+  ORDER_DESC = 2;
+}
+
+enum AggregationType {
+  AGGREGATION_AVG = 0;
+  AGGREGATION_MEDIAN = 1;
+  AGGREGATION_MIN = 2;
+  AGGREGATION_MAX = 3;
+}
+
+// See ListSessionGroups in http_api.md.
+message ListSessionGroupsResponse {
+  repeated SessionGroup session_groups = 1;
+  string next_page_token = 2;
```

Comment:
What is the response when you're at the end of the result set?  Is there any special response?